### PR TITLE
Use static dispatch for Minidump::reader.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

**Comment by frewsxcv** [comment]
Alternative to https://github.com/luser/rust-minidump/pull/5

File: examples/dumpmodules.rs [hunk 1]
```diff
@@ -27,7 +28,7 @@ enum Verbose {
 
 fn print_minidump_modules<T: AsRef<Path>>(path: T,
                                           verbose: Verbose) {
-    match Minidump::read_path(path.as_ref()) {
+    match Minidump::<File>::read_path(path.as_ref()) {
```
Comment (frewsxcv) [inline examples/dumpmodules.rs]: i have no idea why rustc needs to be told explicitly what `T` is here, despite `read_path` explicitly stating what `T` is in the return type

Comment (frewsxcv) [reply examples/dumpmodules.rs]: https://github.com/rust-lang/rust/issues/46988

Comment (frewsxcv) [reply examples/dumpmodules.rs]: nevermind, figured this out

File: minidump-processor/src/processor.rs [hunk 2]
```diff
@@ -85,9 +85,9 @@ pub enum ProcessError {
 /// # }
 /// # fn main() { foo().unwrap() }
 /// ```
-pub fn process_minidump<P>(dump: &mut Minidump,
-                           symbol_provider: &P)
-                           -> Result<ProcessState, ProcessError>
+pub fn process_minidump<T: Readable, P>(dump: &mut Minidump<T>,
```
Comment (luser) [inline minidump-processor/src/processor.rs]: nit: the bounds for `T` should probably go down in the existing `where` clause.

File: src/minidump.rs [hunk 3]
```diff
@@ -45,8 +45,8 @@ use system_info::*;
 /// [read]: struct.Minidump.html#method.read
 /// [read_path]: struct.Minidump.html#method.read_path
 #[allow(dead_code)]
-pub struct Minidump {
-    reader : Box<Readable + 'static>,
+pub struct Minidump<T : Readable> {
+    reader : T,
```
Comment (luser) [inline src/minidump.rs]: FYI if you wind up submitting any other PRs, feel free to fix my weird non-standard formatting here. (I'm not sure where I got the idea that I needed spaces on both sides of the colon.)

File: src/minidump.rs [hunk 4]
```diff
@@ -1291,13 +1293,13 @@ impl Minidump {
     /// the stream data as a specific type.
     ///
     /// [stream]: trait.MinidumpStream.html
-    pub fn get_stream<T: MinidumpStream>(&mut self) -> Result<T, Error> {
-        match self.streams.get_mut(&T::stream_type()) {
+    pub fn get_stream<S: MinidumpStream>(&mut self) -> Result<S, Error> {
```
Comment (luser) [inline src/minidump.rs]: These actually wind up being nicer not using `T` for every single generic type!

**Pull request merged**